### PR TITLE
Prevent Groovy illegal access warnings

### LINKED ISSUE #6725: Gradle 4.10: Illegal reflective access to method java.lang.Object.finalize()
I'm running `gradle -version` and receive the following warning. Browsing the issues here it seems that the current assumption is that with bundled groovy-version this should no happen any longer

### Expected Behavior
Gradle should not spill out warnings the user cannot provide

### Current Behavior
```
 $ gradle --version -s
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/srv/private/Development/gradle-4.10/lib/groovy-all-2.4.15.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

------------------------------------------------------------
Gradle 4.10
------------------------------------------------------------

Build time:   2018-08-27 18:35:06 UTC
Revision:     ee3751ed9f2034effc1f0072c2b2ee74b5dce67d

Kotlin DSL:   1.0-rc-3
Kotlin:       1.2.60
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          10.0.2 (Oracle Corporation 10.0.2+13-Ubuntu-1ubuntu0.18.04.1)
OS:           Linux 4.15.0-33-generic amd64
```

### Steps to Reproduce (for bugs)
Start `gradle --version` in the aforementioned environemnt

### PR DESCRIPTION
**Pull request by oehme**
Start all processes that are likely to use Groovy with the
necessary --add-opens arguments to avoid noisy warnings that
our users can't do anything about.

Fixes #6725

### CODE REVIEW

**Comment by oehme** [comment]
I'm not convinced we should do this for `Test` - The list of illegally accessed packages there is getting longer and longer as I try to run our own test suite.

File: subprojects/base-services/src/main/java/org/gradle/internal/jvm/GroovyJpmsWorkarounds.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class GroovyJpmsWorkarounds {
+    /**
+     * These JVM arguments should be passed to any process that will be using Groovy on Java 9+ to avoid noisy illegal access warnings that the user can't do anything about.
+     */
+    public static final List<String> SUPPRESS_COMMON_GROOVY_WARNINGS = Collections.unmodifiableList(Arrays.asList(
```
Comment (marcphilipp) [inline subprojects/base-services/src/main/java/org/gradle/internal/jvm/GroovyJpmsWorkarounds.java]: Can we use `ImmutableList.of()` here?

Comment (oehme) [reply subprojects/base-services/src/main/java/org/gradle/internal/jvm/GroovyJpmsWorkarounds.java]: Nope, we don't want to load Guava in the command line client.

**Comment by oehme** [comment]
We could even decide to pass `--illegal-access=deny` to the daemon and tell users to add the relevant `--add-opens` if a plugin wants do do illegal accesses. But maybe that's too early and should be done in 6.0, when more libraries out there are fixed.

**Comment by oehme** [comment]
@marcphilipp I've removed the add-opens for `Test` tasks and cut the list of add-opens down to the minimum needed for most Groovy build scripts and GroovyCompile. I've adjusted our own Test tasks to deny illegal access and add the relevant opens clauses. This is what users would have to do as well, with the add-opens depending on their code.

**Comment by oehme** [comment]
Even trying to only open up the things our tests really need proves to be quite a lot - There are many illegal acceses currently hidden behind the one-shot warning: https://builds.gradle.org/viewQueued.html?itemId=16496243

**Comment by marcphilipp** [comment]
Indeed. 😲

**Comment by oehme** [comment]
I've fixed the most obvious one (the TAPI doing an illegal access). CodeNarc is broken because Groovy is broken. The illegal accesses in the embedded executer (which also affects TestKit in debug mode) are harder to get rid of. The illegal accesses in the file tree tests are just in the test code and thus less of a problem, but we should probably still find a better way, e.g. by forking another process to test them.

**Comment by wolfs** [comment]
@oehme I reverted your change since it caused CI breakages: https://builds.gradle.org/viewLog.html?buildId=16582372&buildTypeId=Gradle_Check_PerformanceTestCoordinator
